src/main/java/org/launchcode/techjobs/oo/Job.java

```diff
@@ -17,9 +18,111 @@ public class Job {
     //  other five fields. The second constructor should also call the first in order to initialize
     //  the 'id' field.
 
+    public Job() {
+        id = nextId;
+        nextId++;
+    }
+
+    public Job(String name, Employer employer, Location location, PositionType positionType, CoreCompetency coreCompetency) {
+        this();
+//        if (name.length() < 1) {
+//            name = "Data not available";
+//        }
+//        else {
+//            this.name = name;
+//        }
+        this.name = name;
+        this.employer = employer;
+        this.location = location;
+        this.positionType = positionType;
+        this.coreCompetency = coreCompetency;
+
+    }
+
     // TODO: Add custom equals and hashCode methods. Consider two Job objects "equal" when their id fields
     //  match.
 
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (!(o instanceof Employer)) return false;
+        Employer employer = (Employer) o;
+        return getId() == employer.getId();
+    }
```

Comment:
```suggestion
    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof Job)) return false;
        Job job= (Job) o;
        return getId() == job.getId();
    }
```

Since the `equals` method is now in `Job.java`, you'll need to update the method to check the Job object instead of the Employer object. Once refactored, this passes the `testEqualsMethod()` test!